visualization/app/codeCharta/ui/copyToClipboardButton/util/getFilenamesWithHighestMetric.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { NodeType } from "../../../codeCharta.model"
+import { FileToValue, getFilenamesWithHighestMetrics, updateAttributeMap } from "./getFilenamesWithHighestMetrics"
+
+let MAP: Map<string, FileToValue[]>
+
+beforeEach(() => {
+	MAP = new Map<string, FileToValue[]>([
+		[
+			"mcc",
+			[
+				{ name: "file2", value: 500 },
+				{ name: "file3", value: 400 },
+				{ name: "file1", value: 300 },
+				{ name: "file1", value: 200 },
+				{ name: "file1", value: 100 },
+				{ name: "file1", value: 50 },
+				{ name: "file1", value: 20 },
+				{ name: "file1", value: 15 },
+				{ name: "file1", value: 10 },
+				{ name: "file1", value: 5 }
+			]
+		]
+	])
+})
+
+describe("getFilenamesWithHighestMetrics", () => {
+	it("should give map which has exactly two keys (mcc, rloc)", () => {
+		const resultMap = getFilenamesWithHighestMetrics(BIG_NODE_WITH_TWO_ATTRIBUTES)
+		const mccResults = resultMap.get("mcc")
+		const rlocResults = resultMap.get("rloc")
+
+		expect(resultMap.size).toBe(2)
+		expect(mccResults).toBeTruthy()
+		expect(rlocResults).toBeTruthy()
+	})
+	it("should restrict to at most 10 files for each attribute", () => {
+		const resultMap = getFilenamesWithHighestMetrics(BIG_NODE_WITH_TWO_ATTRIBUTES)
+
+		for (const files of resultMap) {
+			expect(files.length).toBeLessThanOrEqual(10)
+		}
+	})
+	it("should return correct values for rloc and mcc", () => {
```

Comment:
We have some test duplication here: This test asserts everything from first test and more. Can we remove the first test?

Reply:
Yes, thanks! We missed this clean-up when we changed the test input.

Reply:
Resolved with commit [5ea8361](https://github.com/MaibornWolff/codecharta/pull/2942/commits/5ea8361f314b3e7d048f68e80716bdbba4f96041)